src/reporting/claim-reporting-fees-nonforked-markets.js

```diff
@@ -0,0 +1,279 @@
+ *   For all non-finalized markets in current universe where the user has unclaimed fees in the reporting participants:
+ *     Call `Market.disavowCrowdsourcers`
+ *
+ * Once the above has been completed:
+ *   Call `FeeWindow.redeem` on all fee windows in the current universe where the user has unclaimed participation tokens
+ *   For reporting participants of non-forked markets:
+ *     Call `DisputeCrowdsourcer.redeem`
+ *     For initial reporters belonging to non-forked markets:
+ *       If a forked market exists in the current universe:
+ *         If the initial reporter belongs to a finalized market:
+ *           Call `InitialReporter.redeem`
+ *       Else:
+ *         Call `InitialReporter.redeem`
+ *
+ * @param {Object} p Parameters object.
+ * @param {string} p.redeemer Ethereum address attempting to redeem reporting fees, as a hexadecimal string.
+ * @param {Array.<string>} p.feeWindows Array of FeeWindow contract addresses which to claim reporting fees, as hexadecimal strings.
+ * @param {ForkedMarket} p.forkedMarket Object containing information about the Forked Market in which the user has unclaimed fees in the Parent Universe(if there is one).
+ * @param {Array.<NonforkedMarket>} p.nonforkedMarkets Array containing objects with information about the non-Forked Markets in which the user has unclaimed fees.
+ * @param {boolean} p.estimateGas Whether to return gas estimates for the transactions instead of actually making the transactions.
+ * @param {{signer: buffer|function, accountType: string}=} p.meta Authentication metadata for raw transactions.
+ * @param {function} p.onSent Called if/when the transactions are broadcast to the network. (Currently used as a placeholder and not actually used by this function.)
+ * @param {function} p.onSuccess Called if/when all transactions are sealed and confirmed.
+ * @param {function} p.onFailed Called if/when all transactions have been attempted and at least one transaction has failed. Error message shows which transactions succeeded and which ones failed.
+ */
+function claimReportingFeesNonforkedMarkets(p) {
+  var payload = immutableDelete(p, ["redeemer", "feeWindows", "forkedMarket", "nonforkedMarkets", "estimateGas", "onSent", "onSuccess", "onFailed"]);
+  var successfulTransactions = {
+    disavowCrowdsourcers: [],
+    feeWindowRedeem: [],
+    crowdsourcerRedeem: [],
+    initialReporterRedeem: [],
+  };
+  var failedTransactions = {
+    disavowCrowdsourcers: [],
+    feeWindowRedeem: [],
+    crowdsourcerRedeem: [],
+    initialReporterRedeem: [],
+  };
+  var gasEstimates = {
+    disavowCrowdsourcers: [],
+    feeWindowRedeem: [],
+    crowdsourcerRedeem: [],
+    initialReporterRedeem: [],
+    totals: {
+      disavowCrowdsourcers: new BigNumber(0),
+      feeWindowRedeem: new BigNumber(0),
+      crowdsourcerRedeem: new BigNumber(0),
+      initialReporterRedeem: new BigNumber(0),
+      all: new BigNumber(0),
+    },
+  };
+
+  if (p.forkedMarket) {
```

Comment:
This comes down to the question again of how we communicate the need to disavow, will we always return a forkedMarket in the object, or put information in with the market?

Reply:
In the spec doc (https://docs.google.com/document/d/1PBa8F2_ZlD3YTve7Dy9PUGM8h_A_qeJKwRfn1PK6Gy4/edit), I wrote:

```
// The forked market object should always be returned when a forked market exists in the current universe, even if the user does not have any claimable fees in it. This is because `claimReportingFeesNonforkedMarkets` needs to know if a non-finalized forked market exists so it can call Market.disavowCrowdsourcers on non-finalized non-forked markets if the forked market is not finalized.
```
However, in looking through the `getReportingFees` function, I don't think we're actually including it if there are no crowdsourcers/initial reporters that the specified user has unclaimed fees in. Is this something you'd like me to add, or would it be easier for you to do it since you're already modifying the `getReportingFees` code?

Reply:
Right, I added the `needsFork` property to the section of the doc containing the format of the returned data. However, that only applies to reporting participants in the forked market, correct? This function only operates on reporting participants in non-forked markets, so I don't think I can use that.

Reply:
hmm, i suppose we could add it as an attribute to markets?

Reply:
If you want to do it that way, always return forked market details, that's fine with me too. I just thought it might be simpler for your implementation if augur-node tracked the requirements of various markets/participants and what they need

Reply:
I think it makes sense to always return forked market details. The Portfolio: Reporting page will also need to know if a forked market exists in order to determine whether to show the forked market card under the Claim button in the UI.